src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -316,10 +321,14 @@ protected function getConstructor(array &$data, string $class, array &$context,
      * is removed from the context before being returned to avoid side effects
      * when recursively normalizing an object graph.
      *
+     * @param array|bool $allowedAttributes
```

Comment:
why reverting argument's type?

Reply:
this should indeed all be reverted. Child classes are allowed to widen the argument type, so this should not be an issue for ApiPlatform.